src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -627,8 +636,29 @@ trait ChoosesMode {
           routeHistory.rememberRoute(leg.travelPath.linkIds, leg.startTime)
         }
       }
+      // HACKY FIX: The old behavior just chose the first RH_TRANSIT itinerary from all the possible RH_TRANSIT
+      // itineraries produced by R5. But different itineraries may have different combinations of walk/rh for
+      // access/egress, and the egress RH legs in different itineraries may have different starting locations. So we'd
+      // really need to send separate RH requests for each RH_TRANSIT itinerary if we wanted to do it correctly. This
+      // sounds too complicated and costly, so instead we just run a route choice on the RH_TRANSIT itineraries returned
+      // by R5. These don't necessarily have correct wait times, but R5 has been updated to give them appropriate costs.
+      // Once we've chosen the best itinerary we can send requests to the RHM to fill in true costs and wait times
+
+      val rhTransitTrip = if (theRouterResult.itineraries.exists(_.tripClassifier == RIDE_HAIL_TRANSIT)) {
+        val attributesOfIndividual =
+          matsimPlan.getPerson.getCustomAttributes
+            .get("beam-attributes")
+            .asInstanceOf[AttributesOfIndividual]
+
+        modeChoiceCalculator(
+          theRouterResult.itineraries.filter(_.tripClassifier == RIDE_HAIL_TRANSIT).toIndexedSeq,
+          attributesOfIndividual,
+          nextActivity(choosesModeData.personData),
+          Some(currentActivity(choosesModeData.personData)),
+          Some(matsimPlan.getPerson)
+        )
+      } else { None }
```

Comment:
We could filter the RHT itineraries here and send them to `modeChoiceCalculator`. It returns None if the alternatives are empty.